src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -603,18 +603,15 @@ protected static bool Matches(ISymbol? symbol1, ISymbol notNullOriginalUnreduced
         var semanticFacts = state.SemanticFacts;
         var semanticModel = state.SemanticModel;
 
-        return IsInNamespaceOrTypeContext()
-            ? SymbolUsageInfo.Create(GetTypeOrNamespaceUsageInfo())
-            : GetSymbolUsageInfoCommon();
+        var topNameNode = node;
+        while (syntaxFacts.IsQualifiedName(topNameNode.Parent))
+            topNameNode = topNameNode.Parent;
 
-        bool IsInNamespaceOrTypeContext()
-        {
-            var current = node;
-            while (syntaxFacts.IsQualifiedName(current.Parent))
-                current = current.Parent;
+        var isInNamespaceNameContext = syntaxFacts.IsBaseNamespaceDeclaration(topNameNode.Parent);
```

Comment:
>e.Parent [](http://example.com/codeflow?start=88&length=8)

Just making sure you intended to use .Parent here

Reply:
oh, I see, you are checking it directly.

Reply:
yup.  we walk to the top name.  But the namespace decl is the parent *of* that top name.

Reply:
which is different from IsInNamespaceOrTypeContext (which operates on hte top name syntax directly).